plugins/PostProcessingPlugin/scripts/MultiExtColorMix.py

```diff
@@ -0,0 +1,623 @@
+                            elif layer_num == end_layer:
+                                lines = layer.split("\n")
+                                # determine retraction and prime------------------------------------
+                                for line in lines:
+                                    if re.search(" X(\d.*) Y(\d.*) E(\d.*)",line):
+                                        post_prime = False
+                                        break
+                                    if re.search("G1 F(\d*) E(\d.*)", line):
+                                        post_prime = True
+                                        break
+                                pre_lines = data[index-1].split("\n")
+                                for pre_line in reversed(pre_lines):
+                                    if re.search(" X(\d.*) Y(\d.*) E(\d.*)",pre_line):
+                                        pre_ret = False
+                                        break
+                                    if re.search("G1 F(\d*) E(.\d*)", pre_line):
+                                        pre_ret = True
+                                        break
+                                # Add the final line depending on retraction and prime
+                                if park_head:
+                                    if not pre_ret and not post_prime:
+                                        final_string = ParkAndPurge.park_script(end_layer, data, park_x, park_y, purge_amt)[0]       
+                                    elif not pre_ret and post_prime:
+                                        final_string = ParkAndPurge.park_script(end_layer, data, park_x, park_y, purge_amt)[1]
+                                    elif pre_ret and not post_prime:
+                                        final_string = ParkAndPurge.park_script(end_layer, data, park_x, park_y, purge_amt)[2]
+                                    elif pre_ret and post_prime:
+                                        final_string = ParkAndPurge.park_script(end_layer, data, park_x, park_y, purge_amt)[3]
+                                elif not park_head:
+                                    final_string = ""
+                                M164str = "\nT" + str(resume_extruder)
+                                data[L] = data[L].replace(";LAYER:" + str(end_layer), ";LAYER:" + str(end_layer) + M164str + final_string)
+                                break
+                        except:
+                            all
+                    if layer_num == end_layer: break   
+        # Add a reset if there isn't already one in place----------------------------------------------------------
+        if not reset_present:
+            data[len(data)-1] = M163_reset + "T" + str(resume_extruder) + "\n" + data[len(data)-1]
+        return data
+        
+class ParkAndPurge:
+    def park_script(purge_layer: str, data: str, park_x: str, park_y: str, retract_amt: str) -> str:
```

Comment:
This doesn't really need to be in a seperate class but can be a static method on the script class
```suggestion
    @staticmethod
    def park_script(purge_layer: str, data: str, park_x: str, park_y: str, retract_amt: str) -> str:
```

Reply:
Done.